sdk/identity/identity/migration-v1-v2.md

```diff
@@ -0,0 +1,309 @@
+# Migrate from v1 to v2 of @azure/identity
+
+Version 2 of the `@azure/identity` package includes the best parts of version 1, plus several improvements. In this document, we'll expand upon the changes.
+
+## Table of contents
+
+- [Node.js version support](#nodejs-version-support)
+- [Dependency changes](#dependency-changes)
+- [Bug fixes](#bug-fixes)
+- [Errors](#errors)
+- [New credentials](#new-credentials)
+- [Changes to existing credentials](#changes-to-existing-credentials)
+  - [Changes to all credentials in general](#changes-to-all-credentials-in-general)
+  - [Changes to the interactive credentials](#changes-to-the-interactive-credentials)
+  - [Changes to ClientSecretCredential and ClientCertificateCredential](#changes-to-clientsecretcredential-and-clientcertificatecredential)
+  - [Changes to the ManagedIdentityCredential](#changes-to-the-managedidentitycredential)
+  - [Changes to the InteractiveBrowserCredential](#changes-to-the-interactivebrowsercredential)
+  - [Changes to the AzureCliCredential](#changes-to-the-azureclicredential)
+  - [Changes to the DeviceCodeCredential](#changes-to-the-devicecodecredential)
+  - [Changes to the VisualStudioCodeCredential](#changes-to-the-visualstudiocodecredential)
+- [Plugins](#plugins)
+  - [Persistent token cache plugin](#persistent-token-cache-plugin)
+  - [VisualStudioCodeCredential plugin](#visualstudiocodecredential-plugin)
+- [Troubleshooting](#troubleshooting)
+- [Provide feedback](#provide-feedback)
+- [Contributing](#contributing)
+
+## Node.js version support 
+
+While the last `@azure/identity` version to support Node.js 8 is 1.3.0, `@types/node` has been updated to version 12. For more details, read the [support policy](https://github.com/Azure/azure-sdk-for-js/blob/main/SUPPORT.md).
+
+## Dependency changes
+
+Version 2 of Azure Identity for JavaScript no longer includes native dependencies (neither ordinary, peer, nor optional dependencies). Previous distributions of `@azure/identity` included an optional dependency on `keytar`, which caused issues for some users in restrictive environments.
+
+Version 2 of Azure Identity for JavaScript now also depends on the latest available versions of `@azure/msal-common`, `@azure/msal-node`, and `@azure/msal-browser`. Our goal is to always be up-to-date with the MSAL versions.
+
+All credentials provided in version 2 of the Identity package for Node.js now make use of the latest `@azure/msal-node`. `InteractiveBrowserCredential` now uses the latest `@azure/msal-node` on Node.js and the latest `@azure/msal-browser` on the browser.
+
+Relying on the latest versions of MSAL comes with many benefits, but it has some considerations:
```

Comment:
Using the term "considerations" would imply that we are telling customers that they have a choice to make given these considerations. Or that they should hold off on moving to v2 due to these considerations. Where-as we want to encourage customers to upgrade. I would recommend the below change in words

```suggestion
Use of MSAL by the credentials brings the below changes in behavior:
```